sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -165,6 +165,85 @@ namespace sycl::ext::oneapi::experimental::matrix {
+Besides matrix multiply and add, matrices are used in linear and non linear piece-wise operations. Activation functions are an example of element-wise operations. They can be linear like `ReLU` that, for each value `z`, returns the maximum between `z` and zero, or non linear like `Sigmoid` that calculates `1/(1+ exp(-z))`. Quantization that is needed for conversion between low precision types like `int8_t` and `fp32` uses piece-wise operations. For instance, quantized GEMM for `int8_t` is calculated using `A*B + sum_rows_A + sum_cols_B + scalar_zero_point`. `sum_rows_A` and `sum_cols_B` do not operate on elements of the matrix but on pieces: row in `sum_rows_A` and columns in `sum_cols_B`.
+
+We explored multiple options to enable this feature in the matrix interface: 1) Allowing non-restrictive element indexing on the matrix elements would result into slow indexing on the GPU, 2) Operator overloading can represent only element-wise operations and not the operations on pieces (row, column, diagonal, etc) of the matrix. 3) Providing specific functions for these piece-wise operations can resolve some of the functions we know of today like the ones involved in quantization but it is not general to any problem that may occur in the future. 
+
+In order to be able to perform any piece-wise operation in a general and in an efficient way, we provide a mapping conversion function from the matrix domain that is owned by a group of work items to the portion that is owned by a work item. Besides, the WI data to joint matrix mapping coordinates information must be known. 
+
+Nvidia wmma interface added a new member to `fragment` class to designate the WI owned part of the matrix. 
+While this provides fast element indexing on the GPU compared to the non-restrictive option, the user does not know the mapping of the owned data to the original matrix. 
+ However using the `mma` ptx instructions as opposed to the `wmma` ptx instructions the mapping is known. Knowing this mapping is important for the user to implement new operations like sum of rows of a matrix for quantized algorithms.
+
+##### Solution: Explicit conversion with mapping from SIMD to SPMD
+We introduce a new function `get_wi_data` that provides a view of the portion of the matrix that is owned by the current WI.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+template <typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout = matrix_layout::row_major,
+          typename Group = sycl::sub_group>
+struct joint_matrix {
+   joint_matrix(Group sg) {…}   
+   wi_data<T, NumRows, NumCols, Layout, Group> get_wi_data() {
+     return wi_data<T, NumRows, NumCols, Layout, Group>(*this);
+   }
+};
+template <typename T, size_t NumRows, size_t NumCols, matrix_layout Layout, typename Group>
+class wi_data {
+  joint_matrix<T, NumRows, NumCols, Layout, Group> &M;
+public:
+  wi_data(joint_matrix<T, NumRows, NumCols, Layout, Group> &Mat) : M(Mat){}
```

Comment:
I presume we do NOT want application code to construct a `wi_data` (or `wi_element`) directly?  Instead, I presume we want application to call `joint_matrix::get_wi_data` to get the `wi_data`.  If that is the case, these constructors should be private in the implementation, and `joint_matrix` should be a friend, so that it can construct the objects.

The coded synopsis, then, would only list the public member functions:

```
template <typename T, size_t NumRows, size_t NumCols, matrix_layout Layout, typename Group>
class wi_data {
 public:
  size_t length();
  wi_element<T, NumRows, NumCols, Layout, Group> operator[](size_t i);
};
```